libs/shared/ui-components/src/SocialCard/types.ts

```diff
@@ -0,0 +1,33 @@
+// og:url -  The canonical URL of your object that will be used as its permanent ID in the graph,
+// example: https://labs.quansight.org/projects
+// example: https://quansight.com/
+// note: should it include query string params?
+// like https://quansight.com/library?page=2&type=blog (?)
+// url?: string;
+
+// og:image - An image URL which should represent your object within the graph.
+// aspect ratio: The recommended size for an OG Image is 1.91:1. The recommended pixel dimensions of 1200:630 px
+// example: 'https://a.storyblok.com/f/152463/900x472/b289c484e3/qslabs-logo-opengraph-size.png';
+
+// twitter:image for large summary cards
+// 2:1 with minimum dimensions of 300x157 or maximum of 4096x4096 pixels.
+// Images must be less than 5MB in size. JPG, PNG, WEBP and GIF formats are supported.
+// Only the first frame of an animated GIF will be used.
+// SVG is not supported.
+
+// twitter:image for small summary cards
+// aspect ratio of 1:1 with minimum dimensions of 144x144 or maximum of 4096x4096 pixels.
+// The image will be cropped to a square on all platforms.
+// other specification's are similar to large summary cards
+
```

Comment:
I think these comment lines should be placed immediately before the type definition and after the import.

Actually, I would break out these comments and match them to each field in the type definition. But that's just my preference.

Reply:
Agreed, I would also prefer the documentation for each field/prop immediately precede each pro.

Reply:
```suggestion
export type TSocialCardProps = {
  title: string;
  description: string;
  variant: DomainVariant;
 // twitter:image for small summary cards
 // aspect ratio of 1:1 with minimum dimensions of 144x144 or maximum of 4096x4096 pixels.
 // The image will be cropped to a square on all platforms.
 // other specification's are similar to large summary cards
  twitterImage?: string; 
  // og:url -  The canonical URL of your object that will be used as its permanent ID in the graph,
 // example: https://labs.quansight.org/projects
// example: https://quansight.com/
// note: should it include query string params?
// like https://quansight.com/library?page=2&type=blog (?)
  ogImage?: string;
  alt?: string;
 // twitter:image for large summary cards
// 2:1 with minimum dimensions of 300x157 or maximum of 4096x4096 pixels.
// Images must be less than 5MB in size. JPG, PNG, WEBP and GIF formats are supported.
// Only the first frame of an animated GIF will be used.
// SVG is not supported.
  summaryLargeImage?: boolean;
};

```
My concern is adding the comment just before the types makes it more unreadable. But my suggestion is adding a one line highlighted linked that contains all the info that is add and for information that's not included in the link adding it just before.

Also once we have the documentation updating it to the doc link.

Reply:
> My concern is adding the comment just before the types makes it more unreadable.

Hm -- I use VS Code, and the color formatting that VS Code applies makes it easily readable for me -- easy to tell what's comment and what's code -- and it's easier to associate the comments with the code bits when they're interleaved.

I can see the readability question in the absence of color formatting. But, I think the convention is to put the documentation for each thing immediately above that thing.

@gabalafou, thoughts?

>  But my suggestion is adding a one line highlighted linked that contains all the info that is add and for information that's not included in the link adding it just before.

Can you rephrase this? I'm not sure what you mean. What's "highlighted linked"... just linking to the specification, rather than including all the details in the documentation comment?

Reply:
Also, I think this is the wrong piece of documentation for `ogImage`... doesn't `ogImage` become `og:image`, not `og:url`?

```
// og:url -  The canonical URL of your object that will be used as its permanent ID in the graph,
// example: https://labs.quansight.org/projects
  ogImage?: string;
```

Aren't `og:url` and `twitter:url`

Reply:
Yes, I think it's worth both linking to the original specifications *and* repeating the details in the documentation.

Having the information right there will make it faster for people to work with the component; and, the link will help people if the spec ever changes and the component starts behaving strangely.

Reply:
By highlighted link I mean the option you get when you highlight a text and I think it helps because when you open the link it will only highlight the text you want the shared user to focus on

<img width="1439" alt="Screenshot 2023-06-01 at 9 18 41 PM" src="https://github.com/Quansight/Quansight-website/assets/53017492/09bc3597-b2ca-4062-ae27-2be0e4ac5934">

Reply:
Ohh, I see what you mean now.

I would be okay with a URL with a fragment (e.g., `https://example.com/page#go-to-id`) that gets as close to the target content as possible, but I strongly dislike those URLs with highlights. They are *extremely* long, and are especially hard to copy out of GitHub's code views.

Also, I personally find the highlighting more intrusive than helpful. They're also not supported yet by all browsers (e.g., [Firefox](https://connect.mozilla.org/t5/ideas/copy-link-to-highlight/idi-p/16058)).